mlir/lib/Catalyst/Driver/CompilerDriver.cpp

```diff
@@ -0,0 +1,443 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <filesystem>
+#include <list>
+#include <memory>
+
+#include "gml_st/transforms/passes.h"
+#include "mhlo/IR/register.h"
+#include "mhlo/transforms/passes.h"
+#include "mlir/IR/DialectRegistry.h"
+#include "mlir/InitAllDialects.h"
+#include "mlir/InitAllExtensions.h"
+#include "mlir/InitAllPasses.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Target/LLVMIR/Export.h"
+#include "stablehlo/dialect/Register.h"
+#include "llvm/Analysis/CGSCCPassManager.h"
+#include "llvm/Analysis/LoopAnalysisManager.h"
+#include "llvm/IRReader/IRReader.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Support/SourceMgr.h"
+
+#include "Catalyst/Driver/CatalystLLVMTarget.h"
+#include "Catalyst/Driver/CompilerDriver.h"
+#include "Catalyst/Driver/Support.h"
+#include "Catalyst/IR/CatalystDialect.h"
+#include "Catalyst/Transforms/Passes.h"
+#include "Gradient/IR/GradientDialect.h"
+#include "Gradient/Transforms/Passes.h"
+#include "Quantum/IR/QuantumDialect.h"
+#include "Quantum/Transforms/Passes.h"
+
+#include "Enzyme.h"
+#include "PreserveNVVM.h"
+
+using namespace mlir;
+using namespace catalyst;
+namespace fs = std::filesystem;
+
+namespace {
+
+std::string joinPasses(const Pipeline::PassList &passes)
```

Comment:
What do you think of inlining this function?

Reply:
I think we should leave inlining to the compiler.